nautobot/extras/migrations/0116_task_queue_to_job_queue_migration.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 4.2.15 on 2024-08-20 14:58
+
+from django.db import migrations
+
+from nautobot.extras.choices import JobQueueTypeChoices
+
+
+def migrate_task_queues_to_job_queues(apps, schema):
+    # Get the relevant models
+    Job = apps.get_model("extras", "Job")
+    JobQueue = apps.get_model("extras", "JobQueue")
+
+    for job in Job.objects.all():
+        task_queues = job.task_queues
+        # remove duplicates?
+        task_queues = list(set(task_queues))
+        # Go through each task_queue
+        # make or get the corresponding job_queue object and assign it to the job
+        for task_queue in task_queues:
+            job_queue, _ = JobQueue.objects.get_or_create(name=task_queue, queue_type=JobQueueTypeChoices.TYPE_CELERY)
+            job.job_queues.add(job_queue)
+
+    Job.objects.all().update(task_queues=[])
```

Comment:
Should there be a subsequent schema migration to remove the `task_queues` field altogether?

Reply:
Do you mind if we open up a followup issue to remove `task_queues` field and its references in the code? It seems like it's gonna be a lot more work than the current size of the story.

Reply:
https://github.com/nautobot/nautobot/issues/6125